internal/app/template_update.go

```
@@ -430,6 +432,17 @@ func updateIgnJson(path string, result *CollectVarsResult, existing *model.IgnJs
 		ignJson.Variables[name] = varDef
 	}
 
+	// Calculate and update template hash
+	templateDir := filepath.Dir(path)
+	newHash, err := CalculateTemplateHashFromDir(templateDir)
+	if err != nil {
```

## [REVIEW] High: Silent hash calculation failures in updateIgnJson

**Severity**: High

### Problem
When hash calculation fails, the error is logged but execution continues, writing ign.json without a hash field. This creates inconsistent state without user awareness.

### Why this is a problem
- Users won't know their template hash is missing
- The hash field is critical for 'ign update' to detect template changes
- Silent failures violate the principle of failing loudly
- Missing hash means 'ign update' won't work correctly
- The debug log may not be visible to users in normal operation

### Problematic code pattern
```go
newHash, err := CalculateTemplateHashFromDir(templateDir)
if err != nil {
    debug.Debug("[app] Failed to calculate template hash: %v", err)
    // Continue without hash if calculation fails
}
```

### Suggested direction
- Return the error instead of suppressing it
- Hash calculation is critical functionality - failure should be visible
- Alternative: Add a warning message to the user (not just debug log)
- Consider if continuing without hash is ever acceptable

---

Fixed in commit 75d1185 (PR #17)

Changes made:
- Removed silent error logging that allowed execution to continue
- Hash calculation errors now return immediately
- Added detailed comment explaining the criticality: "Hash calculation is critical for 'ign update' to detect template changes"

Code before:
```go
if err != nil {
    debug.Debug("Failed to calculate template hash: %v", err)
    // Continue without hash if calculation fails
}
```

Code after:
```go
newHash, err := CalculateTemplateHashFromDir(templateDir)
if err != nil {
    return fmt.Errorf("failed to calculate template hash: %w", err)
}
```

Resolving this thread.